saga_tools/slurm_run_python.py

```diff
@@ -0,0 +1,225 @@
+
+    @staticmethod
+    def from_parameters(params: Parameters) -> "SlurmPythonRunner":
+        return SlurmPythonRunner(
+            conda_base_path=params.existing_directory("conda_base_path").absolute(),
+            conda_environment=params.string("conda_environment_name"),
+            log_base_directory=params.creatable_directory("log_directory").absolute(),
+        )
+
+    def run_entry_point(
+        self,
+        entry_point_name: str,
+        param_file: Path,
+        *,
+        partition: str,
+        working_directory: Path,
+        num_gpus: int = 0,
+        num_cpus: int = 1,
+        job_name: Optional[str] = None,
+        memory_request: MemoryAmount,
+        slurm_script_path: Optional[Path] = None,
+        echo_template: bool = False,
+    ):
+        slurm_script_directory = slurm_script_path.parent if slurm_script_path else None
+        job_log_directory = self._job_log_directory(job_name)
+        with temppathlib.TmpDirIfNecessary(path=slurm_script_directory) as tmp_dir:
+            if not slurm_script_path:
+                slurm_script_path = tmp_dir.path / f"{job_name}.sbatch"
+
+            # Whether we use an account parameter or a quality of service parameter
+            # depends on whether we are running on a project partition
+            # or one of the available-to-everyone-but-you-can-be-fkilled-at-any-time partitions.
+            if partition in ("scavenge", "ephemeral"):
+                account_or_qos = f"#SBATCH --qos={partition}"
+            else:
+                account_or_qos = f"#SBATCH --account={partition}"
+
+            slurm_template_content = SLURM_BATCH_TEMPLATE.format(
+                partition=partition,
+                account_or_qos=account_or_qos,
+                job_name=job_name,
+                memory_string=self._to_slurm_memory_string(memory_request),
+                num_cpus=num_cpus,
+                num_gpus=num_gpus,
+                stdout_log_path=job_log_directory / f"{job_name}.log",
+                spack_setup_lines="",
+                conda_base_path=self.conda_base_path,
+                conda_environment=self.conda_environment,
+                working_directory=working_directory,
+                entry_point=entry_point_name,
+                param_file=param_file.absolute(),
+            )
+            slurm_script_path.write_text(  # type: ignore
+                slurm_template_content, encoding="utf-8"
+            )
+            if echo_template:
+                print(slurm_template_content)
+            subprocess.run(
+                ["sbatch", str(slurm_script_path.absolute())],  # type: ignore
+                # Raise an exception on failure
+                check=True,
+            )
+
+    def _job_log_directory(self, job_name: str) -> Path:
+        """
+        Gets the directory to write the job logs to.
+
+        This will be `log_base_directory` unless the job name contains */*s.
```

Comment:
I was confused with the `*/*s`, so maybe say explicitly "forward slashes" in just one place; the later examples are clear.